### PR TITLE
Add more PTP features, port-channel support and bugfix

### LINKED ISSUE #538: Enhancement for PTP
**Issue Type**

- Template enhancement
- Role enhancement
- Documentatin enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**

Add following knobs for ptp

```
ptp:
  announce:
    interval: < integer>
  sync-message:
    interval: <-integer>
  delay-req: <-integer>
  role: < master | slave >
```

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by onurgashi**
## Change Summary

This will add more features on interface level to PTP. Additionally we have added support for port-channel configuraitons.
Also fixed a bug with ptp dscp missing default at the end.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #538 

## Component(s) name
eos_cli_config_gen
ethernet-interfaces.j2
port-channel-interfaces.j2
ptp.j2
readme
Molecule test case

## Proposed changes

Interface level documentation has to be fixed later, I have opened issue #631 to track.

## How to test

Molecule tests positive.

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/ptp.yml [hunk 1]
```diff
@@ -13,15 +13,65 @@ ptp:
...
+      transport: ipv4
+# port-channel interfaces
+  Ethernet5:
+    peer: DC1-AGG01
+    peer_interface: Ethernet1
+    peer_type: l2leaf
+    description: DC1-AGG01_Ethernet1
+    channel_group:
+      id: 5
+      mode: active
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/ptp.yml]: missing newline

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 2]
```diff
@@ -724,6 +724,19 @@ port_channel_interfaces:
...
+      enable: < true | false >
+      announce:
+        interval: < integer >
+        timeout: < integer >
+      delay_req: < integer >
+      delay_mechanism: < e2e | p2p >
+      sync_message:
+        interval: < integer >
+      role: < master | dynamic >
+      role: < master | dynamic >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: duplicate line

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 3]
```diff
@@ -836,6 +859,17 @@ ethernet_interfaces:
...
       enable: < true | false >
+      announce:
+        interval: < integer >
+        timeout: < integer >
+      delay_req: < integer >
+      delay_mechanism: < e2e | p2p >
+      sync_message:
+        interval: < integer >
+      role: < master | dynamic >
+      role: < master | dynamic >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: duplicate line

Comment (onurgashi) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Fixed, thanks!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 2 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Fixed, thanks!

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/ptp.yml (hunk 1 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/ptp.yml]: Fixed, thanks!

**Comment by onurgashi** [comment]
Added PTP domain as well, run the molecule test and added documentation for PTP domain. Let me know if any changes are needed.

**Review by titom73 — APPROVED** [review]
Looks good to me!

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Pull request merged**